src/pineko/scale_variations.py

```diff
@@ -6,55 +6,149 @@
 
-    nf is the number of active flavors.
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    delta : int
+        difference between asked order and m
+
+    Returns
+    -------
+    dict(tuple(int))
+        description of all the needed orders
     """
-    return -k_R * m * beta.beta_qcd((2, 0), nf)
-
+    nec_orders = {}
+    for delt in range(delta):
+        nec_orders[(m + delta, 0, delt + 1, 0)] = [
+            (m + de, 0, 0, 0) for de in range(delta - delt)
+        ]
+    return nec_orders
+
+
+def create_svonly_grid(grid, order, new_order, scalefactor):
+    """Create a grid containing only the renormalization scale variations at a given order for a grid."""
+    # Retrieve parameters to create new grid
+    bin_limits = [
+        float(bin) for bin in range(grid.raw.bins() + 1)
+    ]  # +1 because I don't know
+    lumi_grid = [pineappl.lumi.LumiEntry(mylum) for mylum in grid.raw.lumi()]
+    subgrid_params = pineappl.subgrid.SubgridParams()
+    new_order = [pineappl.grid.Order(*new_order)]
+    # create new_grid with same lumi and bin_limits of the original grid but with new_order
+    new_grid = pineappl.grid.Grid.create(
+        lumi_grid, new_order, bin_limits, subgrid_params
+    )
+    # extract the relevant order to rescale from the grid for each lumi and bin
+    grid_orders = [orde.as_tuple() for orde in grid.orders()]
+    order_index = grid_orders.index(order)
+    for lumi_index in range(len(lumi_grid)):
+        for bin_index in range(grid.raw.bins()):
+            extracted_subgrid = grid.subgrid(order_index, bin_index, lumi_index)
+            extracted_subgrid.scale(scalefactor)
+            # Set this subgrid inside the new grid
+            new_grid.set_subgrid(0, bin_index, lumi_index, extracted_subgrid)
+    return new_grid
+
+
+def create_all_necessary_grids(gridpath, order, kR, nf):
+    """Create all the necessary scale variations grids for a certain starting grid."""
+    grid = pineappl.grid.Grid.read(gridpath)
+    grid_orders = [orde.as_tuple() for orde in grid.orders()]
+    first_nonzero_order = grid_orders[0]
+    m_value = first_nonzero_order[0]
+    deltaorder = order[0] - m_value
+    nec_orders = compute_map_of_necessary_orders(m_value, deltaorder)
+    grid_list = []
+    for to_construct_order in nec_orders:
+        for nec_order in nec_orders[to_construct_order]:
+            scalefactor = compute_scalefactor(
+                m_value, nec_order, to_construct_order, nf, kR
+            )
+            grid_list.append(
+                create_svonly_grid(grid, nec_order, to_construct_order, scalefactor)
+            )
+    return grid_list
 
-def NNLO_ren_var(m, k_R, nf):
-    """NNLO ren scale variations depending on the first non vanishing order m.
 
-    k_R is defined as the log(mu_{R}^{2}/ Q^2).
+def construct_ren_sv_grid(grid, max_as, nf):
```

Comment:
```suggestion
def build_ren_sv(grid, max_as, nf):
```
build is more frequent, but to be more consistent with the other functions
```suggestion
def compute_ren_sv(grid, max_as, nf):
```